### ISSUE TITLE
Different effective learning rate reported over gpus

### ISSUE THREAD

**Issue by kuacakuaca** [rwth-i6/returnn#1493]
In my torch 4-gpu distributed training, I observed that the effective learning rate reported across different gpus were not the same, is this an expected behaviour? I used the `newbob_multi_epoch` learning rate schedule. thanks!

```
start epoch 28 global train step 364338 with effective learning rate 0.0002 ...
start epoch 28 global train step 364338 with effective learning rate 0.0002 ...
start epoch 28 global train step 364338 with effective learning rate 0.0002 ...
dev: score 1.098 error 0.297
We have stored models for epochs [22, 23, 24, 25, 26, 27] and keep epochs [24, 25, 26, 27].
We will delete the models of epochs [22, 23].
Deleted 1.0GB.
start epoch 28 global train step 364338 with effective learning rate 0.00018340080864093426 ...
Trained 13484 steps, 1:26:34 elapsed (96.5% computing time)
Trained 13484 steps, 1:27:43 elapsed (95.2% computing time)
Total train loss: ce 1.050 ce_0 1.401 ce_1 1.181 fer 0.305
Trained 13484 steps, 1:27:43 elapsed (95.3% computing time)
Total train loss: ce 1.050 ce_0 1.400 ce_1 1.180 fer 0.304
Trained 13484 steps, 1:27:43 elapsed (95.3% computing time)
Total train loss: ce 1.055 ce_0 1.407 ce_1 1.187 fer 0.305
Total train loss: ce 1.034 ce_0 1.384 ce_1 1.163 fer 0.301
Evaluating dataset 'dev'
start epoch 29 global train step 377822 with effective learning rate 0.0002 ...
start epoch 29 global train step 377822 with effective learning rate 0.0002 ...
start epoch 29 global train step 377822 with effective learning rate 0.0002 ...
dev: score 1.069 error 0.293
5 epochs stored so far and keeping all.
start epoch 29 global train step 377822 with effective learning rate 0.00018340080864093426 ...

```

**Comment by albertz** [comment]
> is this an expected behaviour?

No, this looks like a bug.

**Comment by albertz** [comment]
I think I know the bug: We do eval only on the first worker, and then the `learning_rate_control` of the other workers don't know about the calculated dev/eval scores.

The fix would be sth like: The other workers need to wait until the eval is ready. Then they should maybe reload the LR file, or the main worker should communicate the eval results to the others (dist broadcast).

**Comment by albertz** [comment]
I will try something. Wait a sec.

**Comment by albertz** [comment]
I pushed sth. Can you try?

**Comment by kuacakuaca** [comment]
thanks! i'll try this out.

**Comment by albertz** [comment]
I pushed another small fix.

**Comment by albertz** [comment]
Sorry, there was another small bug. I pushed another small fix.

**Comment by kuacakuaca** [comment]
thanks! I just wanted to report that.

**Comment by kuacakuaca** [comment]
and now all training processes starts after the evaluation is done.

**Comment by albertz** [comment]
> and now all training processes starts after the evaluation is done.

Which is correct, or not?

**Comment by kuacakuaca** [comment]
yes.
